### DISCUSSION TITLE
RuntimeError: CodecError { IO error: `bytes remaining on stream' }

### DISCUSSION THREAD

**Discussion by leowu4ever** [sfu-db/connector-x#362] [Q&A]
Hi community, I got this error where the command tries to pull an entire table which has around 2m rows. Connectorx was able to handle it nicely but suddenly it starts to give me the error below. Can anyone take a look at this? Thanks.

File ~/opt/anaconda3/lib/python3.9/site-packages/connectorx/__init__.py:224, in read_sql(conn, query, return_type, protocol, partition_on, partition_range, partition_num, index_col)
    221 except ModuleNotFoundError:
    222     raise ValueError("You need to install pandas first")
--> 224 result = _read_sql(
    225     conn,
    226     "pandas",
    227     queries=queries,
    228     protocol=protocol,
    229     partition_query=partition_query,
    230 )
    231 df = reconstruct_pandas(result)
    233 if index_col is not None:

RuntimeError: CodecError { IO error: `bytes remaining on stream' }

**Comment by Zevin18** [comment]
I got the same error yesterday, and I added a protocol parameter then it worked, like cx.read_sql(url, query, protocol='cursor').

**Comment by leowu4ever** [comment]
which db do you use? I am using MySQL and it doesn't support passing cursor as the protocol. Only binary and text, neither works.

**Reply by Zevin18** [reply]
Postgres, it supports three protocols (binary, csv and cursor), and only CURSOR works.
